core/lib/libmbedtls/src/tee_lmd_provider.c

```diff
@@ -9,34 +9,195 @@
 #include <crypto/aes-gcm.h>
 #include <crypto/crypto.h>
 #include <kernel/panic.h>
+#include "mbedtls.h"
 #include <stdlib.h>
+#include <string_ext.h>
+#include <string.h>
+#include <utee_defines.h>
+
+#if defined(_CFG_CRYPTO_WITH_HASH) || defined(CFG_CRYPTO_RSA) || \
+	defined(CFG_CRYPTO_HMAC)
+/*
+ * Get mbedtls hash info given a TEE Algorithm "algo"
+ * Return
+ * - mbedtls_md_info_t * in case of success,
+ * - NULL in case of error
+ */
+static const mbedtls_md_info_t *tee_algo_to_mbedtls_hash_info(uint32_t algo)
+{
+	switch (algo) {
+#if defined(CFG_CRYPTO_SHA1)
+	case TEE_ALG_RSASSA_PKCS1_V1_5_SHA1:
+	case TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA1:
+	case TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA1:
+	case TEE_ALG_SHA1:
+	case TEE_ALG_DSA_SHA1:
+	case TEE_ALG_HMAC_SHA1:
+		return mbedtls_md_info_from_string("SHA1");
```

Comment:
Why not return `&mbedtls_sha1_info` directly instead of going through two additional lookup functions.